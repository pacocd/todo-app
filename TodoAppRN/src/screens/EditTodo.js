import React, { Component } from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';
import TodoInput from '../components/TodoInput';
import SquareButton from '../components/SquareButton';

export class EditTodo extends Component {
  static propTypes = {
    navigation: PropTypes.shape({
      state: PropTypes.shape({
        params: PropTypes.shape({
          todo: PropTypes.shape({
            id: PropTypes.number.isRequired,
            user_id: PropTypes.number.isRequired,
            title: PropTypes.string.isRequired,
            completed: PropTypes.bool.isRequired
          }).isRequired
        }).isRequired
      }).isRequired
    }).isRequired
  };

  static navigationOptions = {
    title: 'Edit'
  };

  render() {
    const { todo } = this.props.navigation.state.params;
    return (
      <View style={{ flex: 1, backgroundColor: 'white' }}>
        <TodoInput
          fieldName="user_id"
          value={todo.user_id}
          type="numeric"
          title="User"
        />
        <TodoInput
          fieldName="title"
          value={todo.title}
          type="text"
          title="Title"
        />
        <TodoInput
          fieldName="completed"
          value={todo.completed}
          type="switch"
          title="Completed"
        />
        <SquareButton
          text="Save"
          textColor="white"
          color="#4688f1"
          onPress={() => {}}
        />
      </View>
    );
  }
}

export default EditTodo;
