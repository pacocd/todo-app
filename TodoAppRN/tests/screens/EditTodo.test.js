import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import EditTodo from '../../src/screens/EditTodo';

configure({ adapter: new Adapter() });

describe('when EditTodo is fully rendered', () => {
  const props = {
    navigation: {
      state: {
        params: { todo: { id: 1, title: 'test', user_id: 1, completed: false } }
      }
    }
  };
  const wrapper = shallow(<EditTodo {...props} />);

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
