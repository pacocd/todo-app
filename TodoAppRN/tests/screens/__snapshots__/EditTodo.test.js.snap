// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when EditTodo is fully rendered renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EditTodo
    navigation={
      Object {
        "state": Object {
          "params": Object {
            "todo": Object {
              "completed": false,
              "id": 1,
              "title": "test",
              "user_id": 1,
            },
          },
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Connect(TodoInput)
          fieldName="user_id"
          title="User"
          type="numeric"
          value={1}
        />,
        <Connect(TodoInput)
          fieldName="title"
          title="Title"
          type="text"
          value="test"
        />,
        <Connect(TodoInput)
          fieldName="completed"
          title="Completed"
          type="switch"
          value={false}
        />,
        <SquareButton
          color="#4688f1"
          onPress={[Function]}
          text="Save"
          textColor="white"
        />,
      ],
      "style": Object {
        "backgroundColor": "white",
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "fieldName": "user_id",
          "title": "User",
          "type": "numeric",
          "value": 1,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "fieldName": "title",
          "title": "Title",
          "type": "text",
          "value": "test",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "fieldName": "completed",
          "title": "Completed",
          "type": "switch",
          "value": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "color": "#4688f1",
          "onPress": [Function],
          "text": "Save",
          "textColor": "white",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Connect(TodoInput)
            fieldName="user_id"
            title="User"
            type="numeric"
            value={1}
          />,
          <Connect(TodoInput)
            fieldName="title"
            title="Title"
            type="text"
            value="test"
          />,
          <Connect(TodoInput)
            fieldName="completed"
            title="Completed"
            type="switch"
            value={false}
          />,
          <SquareButton
            color="#4688f1"
            onPress={[Function]}
            text="Save"
            textColor="white"
          />,
        ],
        "style": Object {
          "backgroundColor": "white",
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "fieldName": "user_id",
            "title": "User",
            "type": "numeric",
            "value": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "fieldName": "title",
            "title": "Title",
            "type": "text",
            "value": "test",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "fieldName": "completed",
            "title": "Completed",
            "type": "switch",
            "value": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "color": "#4688f1",
            "onPress": [Function],
            "text": "Save",
            "textColor": "white",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
